src/object/endsequence.hpp

```diff
@@ -30,24 +32,28 @@ class EndSequence : public GameObject
   virtual void draw(DrawingContext& context) override;
 
   void start(); /**< play EndSequence */
-  void stop_tux(); /**< called when Tux has reached his final position */
+  void stop_tux(int player); /**< called when Tux has reached his final position */
   void stop(); /**< stop playing EndSequence, mark it as done playing */
-  bool is_tux_stopped() const; /**< returns true if Tux has reached his final position */
+  bool is_running() const; /**< returns true if the ending cinematic started */
+  bool is_tux_stopped(int player); /**< returns true if Tux has reached his final position */
   bool is_done() const; /**< returns true if EndSequence has finished playing */
   virtual bool is_saveable() const override {
     return false;
   }
 
+  const Controller* get_controller(int player);
```

Comment:
Can this be marked `const` ?

Reply:
It can't, because under the hood it uses the private `get_code_controller()`, which is non-const because it lazily creates the controller. This is useful because it avoids the ending sequence to query the status of all players in the game, but it has the downside of forcing the functions to be non-const.